content/client-lib-development-guide/features.textile

```diff
@@ -682,18 +693,17 @@ h4. ChannelOptions
 * @(TB1)@ options provided when instancing a channel, see "Java ChannelOptions":https://github.com/ably/ably-java/blob/master/src/io/ably/types/ChannelOptions.java as a reference
 * @(TB2)@ The attributes of @ChannelOptions@ consist of:
 ** @(TB2a)@ @encrypted@ boolean - when true, enables automatic encryption & decryption of all messages
-** @(TB2b)@ @cipherParams@ @CipherParams@ - when encrypted is true, the cipher params are required
+** @(TB2b)@ @cipherParams@ - This is required if @encrypted@ is true. It consists of either:
+*** @(TB2b1)@ A @CipherParams@ instance, or
+*** @(TB2b2)@ an options hash (or language equivalent) consisting of the key @key@ with value a key. In this case, the client library should call `getDefaultParams`, passing it the key, to obtain a @CipherParams@ instance
```

Comment:
> `getDefaultParams` is supposed to be a convenience method for the most usual case

@tcard In the proposal above, almost no-one will *have* to call `getDefaultParams` directly, since almost everything that can be done with that can also be done by supplying the things straight to ChannelOptions and letting it call `getDefaultParams` -- the proposal allows your suggestion of alternative constructor for that as a convenience for the common case (eg `ChannelOptions.withCipher("key")`, fourth bullet point), so there's less need for `getDefaultParams` to be convenient.